CMakeLists.txt

```diff
@@ -122,3 +136,39 @@ add_subdirectory(Applications)
 
 # FIXME(Charly) Not sure this is really the greatest idea ever
 add_subdirectory(Plugins)
+
+
+################################################################################
+# add a target to generate API documentation with Doxygen                      #
+################################################################################
+
+find_package(Doxygen)
+if(DOXYGEN_FOUND)
+  # get changeset id
+  find_package(Git)
+  if(GIT_FOUND)
+    execute_process(COMMAND ${GIT_EXECUTABLE} rev-parse HEAD
+      WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+      OUTPUT_VARIABLE GIT_CHANGESET)
```

Comment:
This variable is actually pretty useful to have even outside of doxygen context (e.g. for Version.cpp)

Can you get it out of the if() ?